platform/glfw/test_writer.cpp

```diff
@@ -0,0 +1,187 @@
+#include "test_writer.hpp"
+
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+#include <ghc/filesystem.hpp>
+
+#include <fstream>
+
+using Writer = rapidjson::PrettyWriter<rapidjson::StringBuffer>;
+
+class TestOperation {
+public:
+    virtual ~TestOperation() = default;
+
+    virtual void serialize(Writer& writer) const = 0;
+};
+
+class SetCamera final : public TestOperation {
+public:
+    SetCamera(const mbgl::CameraOptions& camera_) : camera(camera_) {}
+
+    void serialize(Writer& writer) const override {
+        if (camera.zoom) {
+            writer.StartArray();
+            writer.String("setZoom");
+            writer.Double(*camera.zoom);
+            writer.EndArray();
+        }
+
+        if (camera.bearing && *camera.bearing != 0.) {
```

Comment:
we should probably use comparison with epsilon, otherwise bearing `0.000000001` with go in

Reply:
I tested this and it doesn't happen, it was overcomplicating the code. :-

What is happening is pitch `59.99999999999999`. Fixed it by doing `writer.Int(std::round(*camera.pitch));` because GL Native always uses round numbers for pitch.